service/controller/chart/resource/release/create.go

```diff
@@ -103,13 +103,21 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 			return nil
 		}
-
-		if helmclient.IsInvalidManifest(err) {
+		if helmclient.IsValidationFailedError(err) {
 			reason := err.Error()
 			reason = fmt.Sprintf("helm validation error: (%s)", reason)
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("helm release %#q failed, %s", releaseState.Name, reason))
 			addStatusToContext(cc, reason, validationFailedStatus)
 
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			resourcecanceledcontext.SetCanceled(ctx)
+			return nil
+		} else if helmclient.IsInvalidManifest(err) {
```

Comment:
This is needed for some cases, for example when CRD is not there yet.

Reply:
Does the order matter or can we have this error first so they are ordered?

Reply:
Validation error during install starts with `unable to build kubernetes objects from release manifest: error validating data... ` but the same error during upgrade start as `error validating data...`. So I think current ordering is correct.

Reply:
OK makes sense 👍